File: src-ui/src/api/permissions.ts
/**
 * Permissions 配置相关 API
 */

import { invoke } from '@tauri-apps/api/core';
import type { PermissionsConfig } from '../types/tauri';

/**
 * 获取 Permissions 配置
 */
export async function getPermissionsConfig(): Promise<PermissionsConfig> {
  return await invoke<PermissionsConfig>('get_permissions_config');
}

/**
 * 更新 Permissions 配置
 * @param config Permissions 配置
 */
export async function updatePermissionsConfig(config: PermissionsConfig): Promise<void> {
  return await invoke<void>('update_permissions_config', { config });
}

/**
 * 清除 Permissions 配置
 */
export async function clearPermissionsConfig(): Promise<void> {
  return await invoke<void>('clear_permissions_config');
}

File: src-tauri/src/commands/permissions.rs
use crate::models::claude_advanced::PermissionsConfig;
use crate::services::PermissionsConfigService;

/// 读取 Permissions 配置
#[tauri::command]
pub async fn get_permissions_config() -> Result<PermissionsConfig, String> {
    PermissionsConfigService::read_permissions().map_err(|e| e.to_string())
}

/// 更新 Permissions 配置
#[tauri::command]
pub async fn update_permissions_config(config: PermissionsConfig) -> Result<(), String> {
    PermissionsConfigService::write_permissions(&config).map_err(|e| e.to_string())
}

/// 清除 Permissions 配置
#[tauri::command]
pub async fn clear_permissions_config() -> Result<(), String> {
    PermissionsConfigService::clear_permissions().map_err(|e| e.to_string())
}
